### ISSUE TITLE
ShareButton - Accessibility

### ISSUE THREAD

**Issue by poclerson** [OKAMca/stack#304]
- Should use react aria patterns/components
- Should not declare `aria-controls="share-buttons"`
  - Storybook a11y error:
    ![Image](https://github.com/user-attachments/assets/043f698a-18c0-42e6-9d99-bb23bac2fd3e)
  - Component looks like: 
    ![Image](https://github.com/user-attachments/assets/4e1226e6-6de6-4fcc-be00-5375f5be4d11)
  - `aria-controls` is used to indicate which element an interactive element controls. If the share button receives an `aria-controls` attribute, an element should have a corresponding id
